app/server/api-generator/src/main/java/io/syndesis/server/api/generator/openapi/util/SpecificationOptimizer.java

```diff
@@ -47,7 +47,12 @@ private SpecificationOptimizer() {
      */
     public static String minimizeForComponent(final OasDocument openApiDoc) {
         final ObjectNode json = (ObjectNode) Library.writeNode(openApiDoc);
-        json.remove(Arrays.asList("info", "tags", "definitions", "externalDocs"));
+        json.remove(Arrays.asList("info", "tags", "definitions", "responses", "externalDocs"));
+
+        final JsonNode components = json.get("components");
+        if (components != null) {
+            ((ObjectNode)components).remove(Arrays.asList("schemas", "responses", "requestBodies", "examples", "headers", "links", "callbacks"));
```

Comment:
Just to make sure, `headers` is not the global parameters from OpenAPI 2, i.e. it's not shared across operations. The global parameters in OpenAPI 2 were used by the Camel component to include them in the request (for any operation).